librz/include/rz_util/set.h

```diff
@@ -19,14 +19,38 @@ RZ_API bool set_p_contains(SetP *s, const void *u);
 RZ_API void set_p_delete(SetP *s, const void *u);
 RZ_API void set_p_free(SetP *p);
 
-typedef HtUP SetU;
+typedef HtUU SetU;
 
 RZ_API SetU *set_u_new(void);
+RZ_API ut64 set_u_size(SetU *s);
 RZ_API void set_u_add(SetU *p, ut64 u);
 RZ_API bool set_u_contains(SetU *s, ut64 u);
 RZ_API void set_u_delete(SetU *s, ut64 u);
 RZ_API void set_u_free(SetU *p);
 
+typedef struct {
+	st64 ti; ///< Table index
+	ut64 bi; ///< Bucket index
+	ut64 v; ///< Current value of the iteration
+} SetUIter;
+
+RZ_API void advance_set_u_iter(SetU *s, SetUIter *it);
+
+#define set_u_iter_init(iter) SetUIter iter = { 0 }
+#define set_u_iter_reset(iter) \
+	do { \
+		iter.ti = 0; \
+		iter.bi = 0; \
+		iter.v = 0; \
+	} while (0)
```

Comment:
I think we should just have a `_reset` API and always use it, no matter if init or a reset later.

Reply:
Removed the init and changed the unit test in a way, so it can be used as an example.